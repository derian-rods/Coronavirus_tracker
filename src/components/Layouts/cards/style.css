.card{
    border-top: 5px solid#ff9710;
    margin: 1em 1em;
    padding: 1.5em;
    border-radius: 10px;
    background: #FAFAFB;
    width: 10rem;
    box-shadow: 0 10px 10px #00000086;
    display: flex;
    justify-content: center;
}

.card__body{
    text-align: center;
}
.card__title{
    font-weight: bold;
    padding: .2em;
}
.card__text{
    padding-top:1em ;
}
.redcolor{
    color:red;
    margin: .5em;
}
.greencolor{
    margin: .5em;

    color:green;
}


@media screen and (max-width: 1024px){
    .card{
        border-top: 5px solid#ff9710;
        margin: 1em 1em;
        padding: 1.5em;
        border-radius: 10px;
        background: #FAFAFB;
        width: 11rem;
        box-shadow: 0 10px 10px #00000086;
    }
    .card__text{
        padding-top:1em ;
        margin-left: 1em;
    }
}

/* 768 */
@media screen and (max-width: 768px){
    .card{
        border-top: 5px solid#ff9710;
        margin: 1em 1em;
        padding: 1.5em;
        border-radius: 10px;
        background: #FAFAFB;
        width: 20rem;
        box-shadow: 0 10px 10px #00000086;
    }
    .card__text{
        padding-top:1em ;
        margin-left: 1em;
    }
}
@media screen and (max-width: 414px){
    .card{
        margin-left: -5em;
        width: 18rem;
    }
    .card__text{
        padding-top:1em ;
        margin-left: 1em;
    }
}

/* media 411px */
@media screen and (max-width: 411px){
    .card{
        margin-left: 2.2em;
        width: 18rem;
    }
    .card__text{
        padding-top:1em ;
        margin-left: 1em;
    }
}
@media screen and (max-width: 375px){
    .card{
        margin-left: 3em;
        width: 15rem;
    }
    .card__text{
        padding-top:1em ;
        margin-left: 1em;
    }
}
@media screen and (max-width: 360px){
    .card{
        margin-left: 2.2em;
        width: 15rem;
    }
    .card__text{
        padding-top:1em ;
        margin-left: 1em;
    }
}
@media screen and (max-width: 320px){
    .card{
        margin-left: 2em;
        width: 13em;
    }
    .card__text{
        padding-top:1em ;
        margin-left: 1em;
    }
}