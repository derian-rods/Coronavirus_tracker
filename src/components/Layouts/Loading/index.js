import React from 'react'
import './style.css'
export const LoadingScreen = () => {
    return (
        <div className="loading__screend">
            <div className="loading__container">
                <img  className="loading__img" src='https://www.flaticon.com/svg/static/icons/svg/2760/2760528.svg' alt="corona virus tracker"/>
            </div>
        </div>
    )
}
