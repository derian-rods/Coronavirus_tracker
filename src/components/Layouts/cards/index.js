import React from 'react';
import PropTypes from 'prop-types';
import './style.css';
export const Card = ({title, total, increment, icon, icon2, color}) => {
    return (
        <div className="card">
           <div className="card__body">
           <h4 className="card__title">{title}</h4>
            <h3 className={color}>
              {
                  increment === 0 ? (
                    <i className="far fa-smile"></i>
                  ):(

                  <span>+{increment && new Intl.NumberFormat("es-ES").format(increment)} <i className={icon2}></i></span>
                  )
              }
            </h3>
            <span className="card__text">
                {new Intl.NumberFormat("es-ES").format(total)} total <i className={icon}></i>
            </span> <br/>
           </div>  
        </div>
    )
}
Card.propType = {
  title: PropTypes.string.isRequired,
  total: PropTypes.number,
  increment: PropTypes.number,
  icon: PropTypes.string,
  icon2: PropTypes.string,
  color: PropTypes.string
}