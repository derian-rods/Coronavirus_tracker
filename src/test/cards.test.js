import React from 'react';
import '@testing-library/jest-dom';
import { Card } from '../components/Layouts/cards';
import { shallow } from 'enzyme';

describe('test for card component', () => {
    test('should show <Cards /> correctly ', () => {
        const title = 'Card title'
        const wrapper = shallow( <Card  title={title} />)
        expect(wrapper).toMatchSnapshot()
    });
});