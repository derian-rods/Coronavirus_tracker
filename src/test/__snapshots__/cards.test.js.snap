// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for card component should show <Cards /> correctly  1`] = `
<div
  className="card"
>
  <div
    className="card__body"
  >
    <h4
      className="card__title"
    >
      Card title
    </h4>
    <h3>
      <span>
        +
         
        <i />
      </span>
    </h3>
    <span
      className="card__text"
    >
      NaN
       total 
      <i />
    </span>
     
    <br />
  </div>
</div>
`;
